src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs

```diff
@@ -232,6 +321,64 @@ private static void AddWebSocketHeaders(HttpRequestMessage request, string secKe
             {
                 request.Headers.TryAddWithoutValidation(HttpKnownHeaderNames.SecWebSocketProtocol, string.Join(", ", options.RequestedSubProtocols));
             }
+            if (options.DeflateOptions is not null)
+            {
+                request.Headers.TryAddWithoutValidation(HttpKnownHeaderNames.SecWebSocketExtensions, GetDeflateOptions(options.DeflateOptions));
+
+                static string GetWindowBitsString(int value) => value switch
```

Comment:
I guess what @scalablecory meant was just to have `value.ToString(CultureInfo.InvariantCulture)` or to use [FormattableString.Invariant(FormattableString)](https://docs.microsoft.com/en-us/dotnet/api/system.formattablestring.invariant?view=net-5.0), not `switch` 😊